src/betterbib/cli/doi2bibtex.py

```diff
@@ -18,20 +21,50 @@ def _get_version_text():
     )
 
 
+def _create_key_for_entry(entry):
+    with open(
+        os.path.dirname(os.path.realpath(__file__))
+        + "/nltk_english_stop_words.txt",
+        "r",
+    ) as f:
+        nltk_english_stop_words = set(f.readlines())
+    bibtex_key = ""
+    if entry.persons and entry.persons["author"]:
+        bibtex_key += "_".join(entry.persons["author"][0].last()).lower()
+    if entry.fields["year"]:
+        bibtex_key += "_" + str(entry.fields["year"])
+    if entry.fields["title"]:
+        bibtex_key += "_" + slugify.slugify(
+            entry.fields["title"],
+            max_length=8,
+            separator="_",
+            stopwords=nltk_english_stop_words,
+        )
+    if not bibtex_key:
+        bibtex_key = "key"
+    return bibtex_key
+
+
 def main(argv=None):
     parser = _get_parser()
     args = parser.parse_args(argv)
     source = crossref.Crossref()
     entry = source.get_by_doi(args.doi)
 
-    bibtex_key = "key"
+    if args.create_key:
+        bibtex_key = _create_key_for_entry(entry)
+    else:
+        bibtex_key = "key"
```

Comment:
Falling back to previous behaviour if `--create-key` is not passed  to the program call